src/index.js

```diff
@@ -2,10 +2,13 @@
 
 import Config from 'src/shark/config'
 import Client from 'src/shark/client'
-import ServiceToken from 'src/shark/service_token'
 
-import ClientError from 'src/shark/client_error'
-import ServerError from 'src/shark/server_error'
+import BrowserServiceTokenClient from 'src/shark/service_token/node_client'
+import NodeServiceTokenClient from 'src/shark/service_token/browser_client'
+
+import SharkError from 'src/shark/error'
+
+const ServiceToken = Config.nodeProcess ? NodeServiceTokenClient : BrowserServiceTokenClient
```

Comment:
I don't have a really good, useful way to do the entire "We need two different clients for _mostly_ the same thing better", but maybe we should have a `service_token_client.js` that acts as kind of a factory by hiding the decision which client to use.

Reply:
I kind of thought this is the factory.
I can put these lines in an extra file, but I don't think that would change anything at all.